apps/riscv-tests/isa/rv64uv/vfncvt.c

```diff
@@ -6,150 +6,383 @@
+  VCMP_U32(30, v4,         0x4815d5f1,         0xcaedea55,     0xcacb0719,     0x49802397,     0x4ae60cf5,     0xcb053c5b,     0x49032589,     0xca6d1e41,     0x4ae84f61,     0xca0a8ecd,     0xca97de2d,     0xcaa0879b,     0xca96fe33,     0x4a5ab5e1,     0x4ae96c2d,     0xcb0ed65d);
+};
+
+// Simple random test with similar values (masked)
+// The numbers are the same of TEST_CASE1
+void TEST_CASE16(void) {
+  VSET(16, e16, m1);
+  //              908.994, -6788.630, -5789.335, 8054.104, 3947.551, 9596.856, 2474.506, 3094.286, 7684.992, -6850.149, -54.922, 7737.443, 4171.873, 5266.611, 9163.839, 5679.187
+  VLOAD_32(v2,        0x44633fa3,     0xc5d4250b,     0xc5b4eaaf,     0x45fbb0d4,     0x4576b8d0,     0x4615f36d,     0x451aa818,     0x45416494,     0x45f027ef,     0xc5d61131,     0xc25bb026,     0x45f1cb8c,     0x45825efb,     0x45a494e4,     0x460f2f5b,     0x45b1797f);
+  VLOAD_8(v0, 0xAA, 0xAA);
+  VCLEAR(v4);
+  asm volatile("vfncvt.rod.f.f.w v4, v2, v0.t");
+  //              0.000, -6788.000, 0.000, 8056.000, 0.000, 9600.000, 0.000, 3094.000, 0.000, -6852.000, 0.000, 7736.000, 0.000, 5268.000, 0.000, 5680.000
+  VCMP_U16(31, v4,     0x0,     0xeea1,     0x0,     0x6fdd,     0x0,     0x70af,     0x0,     0x6a0b,     0x0,     0xeeb1,     0x0,     0x6f8f,     0x0,     0x6d25,     0x0,     0x6d8b);
+
+  VSET(16, e32, m1);
+  //              153431.766, -7796010.957, -6652812.196, 1049714.758, 7538298.328, -8731739.480, 537176.622, -3884944.157, 7612336.042, -2270131.404, -4976406.726, -5260237.163, -4947737.810, 3583352.355, 7648790.331, -9360989.228
+  VLOAD_64(v2,        0x4102babe20435c2f,     0xc15dbd4abd4015a9,     0xc15960e30c871450,     0x41300472c1efbd9f,     0x415cc19e94ffb79b,     0xc160a78b6f5bcd25,     0x412064b13e931aa9,     0xc14da3c81425b914,     0x415d09ec02a8cc93,     0xc14151d9b3c1ecaf,     0xc152fbc5ae718384,     0xc15410f34a6ddb48,     0xc152dfc673d9ba53,     0x414b56bc2d765fad,     0x415d2d85952e8398,     0xc161dacba74d791e);
+  VLOAD_8(v0, 0xAA, 0xAA);
+  VCLEAR(v4);
+  asm volatile("vfncvt.rod.f.f.w v4, v2, v0.t");
+  //              0.000, -7796011.000, 0.000, 1049714.750, 0.000, -8731739.000, 0.000, -3884944.250, 0.000, -2270131.500, 0.000, -5260237.000, 0.000, 3583352.250, 0.000, -9360989.000
+  VCMP_U32(32, v4,     0x0,     0xcaedea55,     0x0,     0x49802397,     0x0,     0xcb053c5b,     0x0,     0xca6d1e41,     0x0,     0xca0a8ecd,     0x0,     0xcaa0879b,     0x0,     0x4a5ab5e1,     0x0,     0xcb0ed65d);
+};
+
+int main(void) {
   enable_vec();
   enable_fp();
+
   TEST_CASE1();
   TEST_CASE2();
+
   TEST_CASE3();
   TEST_CASE4();
+
   TEST_CASE5();
   TEST_CASE6();
+
   TEST_CASE7();
   TEST_CASE8();
+
   TEST_CASE9();
   TEST_CASE10();
+
   TEST_CASE11();
   TEST_CASE12();
+
   TEST_CASE13();
   TEST_CASE14();
+
+/*
+vfncvt.rod.f.f is not supported yet
```

Comment:
What is missing for supporting this instruction?

Reply:
The FPU (fpnew) does not support rounding towards odd. Don't think it's hard to implement, especially if we need it only for this operation